File: benchmarks/python/ridgefcn.py
import numpy as np

def ridgefcn(x: np.ndarray, d: float = 1, alpha: float = 0.5) -> np.ndarray:
    """
    Computes the value of the Ridge benchmark function.

    Parameters:
    -----------
    x: np.ndarray, shape=(M, N)
        Matrix of size M-by-N containing the input points.
    d: float, optional (default=1)
        Contribution coefficient of the sphere component of the function.
    alpha: float, optional (default=0.5)
        Power of the sphere component of the function.

    Returns:
    --------
    scores: np.ndarray, shape=(M, 1)
        Vector of size M-by-1 containing the function value for each row of X.

    Author: Mazhar Ansari Ardeh
    Please forward any comments or bug reports to mazhar.ansari.ardeh at
    Google's e-mail service or feel free to kindly modify the repository.
    """
    x1 = x[:, 0]
    scores = x1 + d * (np.sum(x[:, 1:]**2, axis=1) ** alpha)
    return scores

File: src/matlab/ridgefcn.m
% Computes the value of the Ridge benchmark function.
% SCORES = RIDGEFCN(X) computes the value of the Ridge function at point X.
% RIDGEFCN accepts a matrix of size M-by-N and returns a vetor SCORES of
% size M-by-1 in which each row contains the function value for the
% corresponding row of X.
% SCORES = RIDGEFCN(X, D) specifies contribution coefficient of the sphere
% component of the function.
% SCORES = RIDGEFCN(X, D, ALPHA) specifies power of the sphere component of
% the function.
%
% For more information, please visit:
% benchmarkfcns.info/doc/ridgefcn
%
% Author: Mazhar Ansari Ardeh
% Please forward any comments or bug reports to mazhar.ansari.ardeh at
% Google's e-mail service or feel free to kindly modify the repository.
function scores = ridgefcn(x, d, alpha)

    if nargin < 3
        alpha = 0.5;
    end
    if nargin < 2
        d = 1;
    end

    x1 = x(:, 1);
    scores = x1 + d * (sum(x(:, 2:end).^2, 2) .^ alpha);
end
